src/_pytest/config/__init__.py

```diff
@@ -1077,7 +1081,7 @@ def __init__(
         self._inicache: dict[str, Any] = {}
         self._override_ini: Sequence[str] = ()
         self._opt2dest: dict[str, str] = {}
-        self._cleanup: list[Callable[[], None]] = []
+        self._exit_stack = contextlib.ExitStack()
```

Comment:
How about calling it "cleanup_stack" to better convey its purpose?

```suggestion
        self._cleanup_stack = contextlib.ExitStack()
```